js_modules/dagit/packages/core/src/runs/RunTags.tsx

```diff
@@ -1,32 +1,47 @@
 import {Box} from '@dagster-io/ui';
 import * as React from 'react';
 
-import {RunTag} from './RunTag';
-import {RunFilterToken} from './RunsFilterInput';
+import {SharedToaster} from '../app/DomUtils';
+import {useCopyToClipboard} from '../app/browser';
 
-interface RunTagType {
-  key: string;
-  value: string;
-}
+import {RunTag, TagType} from './RunTag';
+import {RunFilterToken} from './RunsFilterInput';
 
 export const RunTags: React.FC<{
-  tags: RunTagType[];
+  tags: TagType[];
   mode: string | null;
   onSetFilter?: (search: RunFilterToken[]) => void;
 }> = React.memo(({tags, onSetFilter, mode}) => {
+  const copy = useCopyToClipboard();
+
   if (!tags.length) {
     return null;
   }
 
-  const onClick = (tag: RunTagType) => {
-    onSetFilter && onSetFilter([{token: 'tag', value: `${tag.key}=${tag.value}`}]);
-  };
+  const actions = [
```

Comment:
You could do 
```
const actions = useMemo(() => {
    const actions = [
      {
        label: 'Copy tag',
        onClick(tag: TagType) {
          copy(`${tag.key}:${tag.value}`);
          SharedToaster.show({intent: 'success', message: 'Copied tag!'});
        },
      },
    ];

    if (onSetFilter) {
      actions.push({
        label: 'Add tag to filter',
        onClick(tag: TagType) {
          onSetFilter([{token: 'tag', value: `${tag.key}=${tag.value}`}]);
        },
      });
    }
    return actions;
  }, []);
```

Not a big deal its a tradeoff between slightly extra memory or slightly longer re-renders. In the future though React will do this for you automatically https://www.youtube.com/watch?v=lGEMwh32soc

Reply:
Oh this is super interesting - thanks for the video link!

Reply:
Yeah, I skipped doing a memo on this since I'd need `onSetFilter` as a dep and I don't have any guarantee that it's a memoized function. I guess since sometimes it's not there at all, it probably couldn't hurt to do it anyway.